tracer/src/Datadog.Trace/AppSec/Security.cs

```diff
@@ -630,38 +630,29 @@ internal bool IsMetaStructSupported()
             return _spanMetaStructs;
         }
 
-        internal void UpdateActiveAddresses()
+        private void UpdateActiveAddresses()
         {
-            // So far, RASP is the only one that uses this
-            if (_settings.RaspEnabled && _waf?.IsKnowAddressesSuported() == true)
+            if (_waf?.IsKnowAddressesSuported() is true)
             {
                 var addresses = _waf.GetKnownAddresses();
                 Log.Debug("Updating WAF active addresses to {Addresses}", addresses);
-                _activeAddresses = addresses is null ? null : new HashSet<string>(addresses);
+                _activeAddresses = [..addresses];
```

Comment:
Nice 👍 but can addresses still be null? If so you'll still need a similar pattern, right?

Reply:
addresses shouldn't be null as per https://github.com/DataDog/dd-trace-dotnet/blob/3e1be0516580fc23cf5b14b004961da97094d6a5/tracer/src/Datadog.Trace/AppSec/Waf/NativeBindings/WafLibraryInvoker.cs#L245